src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'blue' | 'green';
+  getArea: () => number;
 }
 
 export class Triangle implements Figure {
+  shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const biggestSide = Math.max(a, b, c);
+    const isIncorrectSides = biggestSide >= a + b + c - biggestSide;
+
+    if (a <= 0 || b <= 0 || c <= 0 || isIncorrectSides) {
```

Comment:
you can create function to check if each side greater than 0 and then just use it in other functions